manifests/crd.yaml

```diff
@@ -1,68 +1,113 @@
+
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    "api-approved.kubernetes.io": "https://github.com/kubernetes/enhancements/pull/1870"
+    controller-gen.kubebuilder.io/version: v0.6.0
+  creationTimestamp: null
   name: noderesourcetopologies.topology.node.k8s.io
 spec:
   group: topology.node.k8s.io
-  versions:
-    - name: v1alpha1
-      served: true
-      storage: true
-      schema:
-        openAPIV3Schema:
-          type: object
-          properties:
-            topologyPolicies:
-              type: array
-              items:
-                type: string
-            zones:
-              type: array
-              items:
-                type: object
-                properties:
-                  name:
-                    type: string
-                  type:
-                    type: string
-                  parent:
-                    type: string
-                  resources:
-                    type: array
-                    items:
-                      type: object
-                      properties:
-                        name:
-                          type: string
-                        capacity:
-                          x-kubernetes-int-or-string: true
-                        allocatable:
-                          x-kubernetes-int-or-string: true
-                  costs:
-                    type: array
-                    items:
-                      type: object
-                      properties:
-                        name:
-                          type: string
-                        value:
-                          type: integer
-                  attributes:
-                    type: array
-                    items:
-                      type: object
-                      properties:
-                        name:
-                          type: string
-                        value:
-                          type: string
-  scope: Namespaced
   names:
+    kind: NodeResourceTopology
+    listKind: NodeResourceTopologyList
     plural: noderesourcetopologies
     singular: noderesourcetopology
-    kind: NodeResourceTopology
-    shortNames:
-      - node-res-topo
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: NodeResourceTopology is a specification for a Foo resource
```

Comment:
I'm not sure about Foo ;) maybe better to replace it to:
is a specification for a hardware resource

Reply:
Hmm, these come directly from the code comments. Added/updated comments for all the exported api types